Something seems to be flaky with this test. In the previous run it was `Reserved != Delivered`, now it's `Partially Delivered != Delivered`. I have no knowledge of what makes this test proven flaky and whether this PR surfaces it (and if so: how?) ...

```console
======================================================================
 FAIL  test_stock_reservation_against_sales_order (erpnext.selling.doctype.sales_order.test_sales_order.TestSalesOrder.test_stock_reservation_against_sales_order)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
    args = (<erpnext.selling.doctype.sales_order.test_sales_order.TestSalesOrder testMethod=test_stock_reservation_against_sales_order>,)
    func = <function TestSalesOrder.test_stock_reservation_against_sales_order at 0x7f243fe756c0>
    kwds = {}
    self = <contextlib._GeneratorContextManager object at 0x7f244050bdd0>
  File "/home/runner/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/test_sales_order.py", line 1912, in test_stock_reservation_against_sales_order
    self.assertEqual(sre_detail.status, "Delivered")
    cancel_stock_reservation_entries = <function cancel_stock_reservation_entries at 0x7f2442d7cae0>
    create_items = <function create_items at 0x7f243fa05c60>
    create_material_receipt = <function create_material_receipt at 0x7f243fa84360>
    dn1 = <DeliveryNote: MAT-DN-2023-00017 docstatus=1>
    dn2 = <DeliveryNote: MAT-DN-2023-00018 docstatus=1>
    get_sre_reserved_qty_details_for_voucher = <function get_sre_reserved_qty_details_for_voucher at 0x7f2442d7c720>
    get_stock_reservation_entries_for_voucher = <function get_stock_reservation_entries_for_voucher at 0x7f2442d7c540>
    has_reserved_stock = <function has_reserved_stock at 0x7f2442d7c860>
    item = <SalesOrderItem: 8df442e336 docstatus=1 parent=SAL-ORD-2023-00054>
    item_code = 'SR Batch and Serial Item 1'
    item_list = [{'item_code': 'SR Item 1', 'warehouse': '_Test Warehouse - _TC', 'qty': 17.0, 'uom': 'Nos', 'rate': 21, 'doctype': 'Sales Order Item'}, {'item_code': 'SR Item 2', 'warehouse': '_Test Warehouse - _TC', 'qty': 46.54, 'uom': 'Kg', 'rate': 68, 'doctype': 'Sales Order Item'}, {'item_code': 'SR Batch Item 1', 'warehouse': '_Test Warehouse - _TC', 'qty': 58.0, 'uom': 'Nos', 'rate': 111, 'doctype': 'Sales Order Item'}, {'item_code': 'SR Batch Item 2', 'warehouse': '_Test Warehouse - _TC', 'qty': 90.127, 'uom': 'Kg', 'rate': 20, 'doctype': 'Sales Order Item'}, {'item_code': 'SR Serial Item 1', 'warehouse': '_Test Warehouse - _TC', 'qty': 84.0, 'uom': 'Nos', 'rate': 117, 'doctype': 'Sales Order Item'}, {'item_code': 'SR Batch and Serial Item 1', 'warehouse': '_Test Warehouse - _TC', 'qty': 98.0, 'uom': 'Nos', 'rate': 56, 'doctype': 'Sales Order Item'}]
    items_details = {'SR Item 1': <Item: SR Item 1>, 'SR Item 2': <Item: SR Item 2>, 'SR Batch Item 1': <Item: SR Batch Item 1>, 'SR Batch Item 2': <Item: SR Batch Item 2>, 'SR Serial Item 1': <Item: SR Serial Item 1>, 'SR Batch and Serial Item 1': <Item: SR Batch and Serial Item 1>}
    properties = <Item: SR Batch and Serial Item 1>
    randint = <bound method Random.randint of <random.Random object at 0x5631698d9c40>>
    reserved_qty = 98.0
    reserved_qty_details = {'4c908125b1': 17.0, '5c1a9bd7d9': 90.127, '631d625a0f': 46.54, '752ad9a5ec': 98.0, '8d20bb9b63': 58.0, 'bb6c07cf84': 84.0}
    se = <StockEntry: MAT-STE-2023-00018 docstatus=2>
    self = <erpnext.selling.doctype.sales_order.test_sales_order.TestSalesOrder testMethod=test_stock_reservation_against_sales_order>
    so = <SalesOrder: SAL-ORD-2023-00054 docstatus=1>
    sre_detail = {'status': 'Partially Delivered', 'reserved_qty': 46.54, 'delivered_qty': 46.54}
    sre_details = [{'status': 'Partially Delivered', 'reserved_qty': 46.54, 'delivered_qty': 46.54}]
    stock_uom = 'Nos'
    uniform = <bound method Random.uniform of <random.Random object at 0x5631698d9c40>>
    warehouse = '_Test Warehouse - _TC'
AssertionError: 'Partially Delivered' != 'Delivered'
- Partially Delivered
+ Delivered
```

--- a/erpnext/stock/doctype/delivery_trip/delivery_trip.json
+++ b/erpnext/stock/doctype/delivery_trip/delivery_trip.json
@@ -66,8 +66,7 @@
    "fieldname": "driver",
    "fieldtype": "Link",
    "label": "Driver",
-   "options": "Driver",
-   "reqd": 1
+   "options": "Driver"
   },
   {
    "fetch_from": "driver.full_name",
@@ -189,10 +188,11 @@
  ],
  "is_submittable": 1,
  "links": [],
- "modified": "2021-04-30 21:21:36.610142",
+ "modified": "2023-06-27 11:22:27.927637",
  "modified_by": "Administrator",
  "module": "Stock",
  "name": "Delivery Trip",
+ "naming_rule": "By \"Naming Series\" field",
  "owner": "Administrator",
  "permissions": [
   {
@@ -228,5 +228,6 @@
  ],
  "sort_field": "modified",
  "sort_order": "DESC",
+ "states": [],
  "title_field": "driver_name"
 }

--- a/erpnext/stock/doctype/delivery_trip/delivery_trip.py
+++ b/erpnext/stock/doctype/delivery_trip/delivery_trip.py
@@ -24,6 +24,9 @@ class DeliveryTrip(Document):
 		)
 
 	def validate(self):
+		if self._action == "submit" and not self.driver:
+			frappe.throw(_("A driver must be set to submit."))
+
 		self.validate_stop_addresses()
 
 	def on_submit(self):

